lib/shopify_theme/api_checker.rb

```diff
@@ -0,0 +1,39 @@
+module ShopifyTheme
+  class APIChecker
+    class APIResult
+      def initialize(http_response)
+        @response = http_response
+      end
+
+      def accessed_api?
+        response.code == 200
+      end
+
+      def cannot_access_api?
+        !accessed_api?
+      end
+
+      def invalid_config?
+        response.code == 401
+      end
+
+      def api_down?
+        (500..599).include?(response.code)
+      end
+
+      private
+      attr_reader :response
+    end
+
+    def initialize(client)
+      @client = client
+    end
+
+    def test_connectivity
+      return APIResult.new(client.get_index)
+    end
+
+    private
+    attr_reader :client
```

Comment:
Yup. But `attr_reader` is usually placed near the top of the class, and doesn't need to be located after `private` . I'd move it up.